Sources/Foundation/FileManager+POSIX.swift

```diff
@@ -470,7 +470,23 @@ extension FileManager {
 
         return self.string(withFileSystemRepresentation: buf, length: Int(len))
     }
-    
+        
+    internal func _recursiveDestinationOfSymbolicLink(atPath path: String) throws -> String {
+        // Throw error if path is not a symbolic link:
+        let path = try _destinationOfSymbolicLink(atPath: path)
+        
+        let bufSize = Int(PATH_MAX + 1)
+        var buf = [Int8](repeating: 0, count: bufSize)
```

Comment:
Might be nice to avoid the double zero'ing that @valeriyvan has been cleaning up

Reply:
@compnerd @valeriyvan How should I implement it if `realpath` wants a `UnsafeMutablePointer<Int8>?` instead of a `UnsafeMutableBufferPointer<Int8>`?

```swift
internal func _destinationOfSymbolicLink(atPath path: String) throws -> String {
    let bufSize = Int(PATH_MAX + 1)
    var buffer = try [Int8](unsafeUninitializedCapacity: bufSize) { buffer, initializedCount in
        let len = try _fileSystemRepresentation(withPath: path) {
            readlink($0, buffer, initializedCount)  // error: Cannot convert value of type 'UnsafeMutableBufferPointer<Int8>' to expected argument type 'UnsafeMutablePointer<Int8>?'
        }
        if len < 0 {
            throw _NSErrorWithErrno(errno, reading: true, path: path)
        }
    }
    return self.string(withFileSystemRepresentation: buffer, length: buffer.count)
}
```

Reply:
`buffer.baseAddress` should give you what you need.

Reply:
Thanks! Should be fixed now.